ENIGMAsystem/SHELL/Graphics_Systems/None/fillin.h

```diff
@@ -0,0 +1,296 @@
+/** Copyright (C) 2017 Faissal I. Bensefia
+*** Copyright (C) 2008-2013 Robert B. Colton, Adriano Tumminelli
+*** Copyright (C) 2014 Seth N. Hetu
+***
+*** This file is a part of the ENIGMA Development Environment.
+***
+*** ENIGMA is free software: you can redistribute it and/or modify it under the
+*** terms of the GNU General Public License as published by the Free Software
+*** Foundation, version 3 of the license or any later version.
+***
+*** This application and its source code is distributed AS-IS, WITHOUT ANY
+*** WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+*** FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+*** details.
+***
+*** You should have received a copy of the GNU General Public License along
+*** with this code. If not, see <http://www.gnu.org/licenses/>
+**/
+#ifndef GS_NONE_FILLIN
+#define GS_NONE_FILLIN
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <stdlib.h>     /* malloc, free, rand */
+#include <unordered_map>
+#include <math.h>
+
+using namespace std;
+#include <algorithm>
+#include <floatcomp.h>
+#include <vector>
+#include <limits>
+#include <map>
+#include <list>
+#include <stack>
+#include "Universal_System/estring.h"
+
+namespace enigma
+{
+	void graphicssystem_initialize();
+	void load_tiles();
+	void delete_tiles();
+	void rebuild_tile_layer(int layer_depth);
+
+	int graphics_create_texture(unsigned width, unsigned height, unsigned fullwidth, unsigned fullheight, void* pxdata, bool mipmap);
+	int graphics_duplicate_texture(int tex, bool mipmap);
+	void graphics_copy_texture(int source, int destination, int x, int y);
+	void graphics_copy_texture_part(int source, int destination, int xoff, int yoff, int w, int h, int x, int y);
+	void graphics_replace_texture_alpha_from_texture(int tex, int copy_tex);
+	void graphics_delete_texture(int texid);
+	unsigned char* graphics_get_texture_pixeldata(unsigned texture, unsigned* fullwidth, unsigned* fullheight);
+	void graphics_samplers_apply();
+
+	bool fill_complex_polygon(const std::list<PolyVertex>& vertices, int defaultColor, bool allowHoles);
+}
+
+namespace enigma_user
+{
+	string draw_get_graphics_error();
+	unsigned vertex_format_create();
+
+	void vertex_format_destroy(int id);
```

Comment:
Are there not headers for these under General?  I thought we moved most of the API there.

Reply:
Only vertex_format_destroy, vertex_format_create and vertex_format_exists, I've removed them now.